econtext/src/main/java/com/adyen/checkout/econtext/EContextComponentProvider.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ozgur on 31/1/2023.
+ */
+
+package com.adyen.checkout.econtext
+
+import android.app.Application
+import android.os.Bundle
+import androidx.annotation.RestrictTo
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.savedstate.SavedStateRegistryOwner
+import com.adyen.checkout.action.ActionHandlingComponent
+import com.adyen.checkout.action.DefaultActionHandlingComponent
+import com.adyen.checkout.action.GenericActionComponentProvider
+import com.adyen.checkout.action.GenericActionDelegate
+import com.adyen.checkout.components.PaymentComponentProvider
+import com.adyen.checkout.components.analytics.AnalyticsMapper
+import com.adyen.checkout.components.analytics.AnalyticsSource
+import com.adyen.checkout.components.analytics.DefaultAnalyticsRepository
+import com.adyen.checkout.components.api.AnalyticsService
+import com.adyen.checkout.components.base.ButtonComponentParamsMapper
+import com.adyen.checkout.components.base.ComponentParams
+import com.adyen.checkout.components.base.lifecycle.get
+import com.adyen.checkout.components.base.lifecycle.viewModelFactory
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.components.model.payments.request.EContextPaymentMethod
+import com.adyen.checkout.components.repository.PaymentObserverRepository
+import com.adyen.checkout.components.ui.SubmitHandler
+import com.adyen.checkout.core.api.HttpClientFactory
+import com.adyen.checkout.core.exception.ComponentException
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+abstract class EContextComponentProvider<
+    ComponentT : EContextComponent<PaymentMethodT>,
+    ConfigurationT : EContextConfiguration,
+    PaymentMethodT : EContextPaymentMethod>(
+    private val componentClass: Class<ComponentT>,
+    overrideComponentParams: ComponentParams? = null,
+) : PaymentComponentProvider<ComponentT, ConfigurationT> {
```

Comment:
Maybe `SessionPaymentComponentProvider` should be used or were you planning on adding sessions later?